src/main/java/apoc/bolt/Bolt.java

```diff
@@ -0,0 +1,160 @@
+import org.neo4j.driver.v1.summary.SummaryCounters;
+import org.neo4j.driver.v1.types.Node;
+import org.neo4j.driver.v1.types.Path;
+import org.neo4j.driver.v1.types.Relationship;
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Label;
+import org.neo4j.graphdb.RelationshipType;
+import org.neo4j.procedure.Context;
+import org.neo4j.procedure.Name;
+import org.neo4j.procedure.Procedure;
+
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static apoc.util.MapUtil.map;
+
+/**
+ * @author AgileLARUS
+ * @since 29.08.17
+ */
+public class Bolt {
+
+    @Context
+    public GraphDatabaseService db;
+
+    @Procedure()
+    @Description("apoc.bolt.load(url-or-key, statement, params, config) - access to other databases via bolt for read")
+    public Stream<RowResult> load(@Name("url") String url, @Name("statement") String statement, @Name(value = "params", defaultValue = "{}") Map<String, Object> params, @Name(value = "config", defaultValue = "{}") Map<String, Object> config) throws URISyntaxException {
+
+        if (params == null) params = Collections.emptyMap();
+        if (config == null) config = Collections.emptyMap();
+        boolean virtual = (boolean) config.getOrDefault("virtual", false);
+        boolean addStatistics = (boolean) config.getOrDefault("statistics", false);
+        UriResolver uri = new UriResolver(url, "bolt");
+        uri.initialize();
+        Driver driver = null;
+        Session session = null;
+        try {
+            driver = GraphDatabase.driver(uri.getUrlDriver(), AuthTokens.basic(uri.getUser(), uri.getPassword()));
+            session = driver.session();
+            if (addStatistics)
+                return Stream.of(new RowResult(toMap(runStatement(statement, session, params).summary().counters())));
+
+            return getRowResultStream(virtual, session, params, statement);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        } finally {
+            session.close();
+            driver.close();
+        }
+    }
+
+    @Procedure()
+    @Description("apoc.bolt.execute(url-or-key, statement, params, config) - access to other databases via bolt for read")
+    public Stream<RowResult> execute(@Name("url") String url, @Name("statement") String statement, @Name(value = "params", defaultValue = "{}") Map<String, Object> params, @Name(value = "config", defaultValue = "{}") Map<String, Object> config) throws URISyntaxException {
+        Map<String, Object> configuration =  new HashMap<>(config);
+        configuration.remove("virtual");
+        return load(url, statement, params, configuration);
+    }
+
+    private StatementResult runStatement(@Name("statement") String statement, Session session, Map<String, Object> finalParams) {
+        return session.writeTransaction((Transaction tx) -> tx.run(statement, finalParams));
```

Comment:
the idea was that "load" uses readOnlyTx, so that e.g. if you have a user that cannot write, or if you want your queries to go to the whole cluster, you can use apoc.bolt.load
and execute uses a write-tx.